backend/app/api/src/crons/clear-excel-cache.ts

```diff
@@ -0,0 +1,92 @@
+import fs from "fs/promises";
+
+const msIn22Hours = 79200000;
+let lastExcelClear: number | null = null;
+
+export async function clearExcelCache() {
+    const now = new Date();
+
+    const didClear = await clearExcelCacheHelper({
+        lastExcelClear,
+        now,
+        cachePath: STAMHOOFD.CACHE_PATH,
+        environment: STAMHOOFD.environment
+    });
+
+    if(didClear) {
+        lastExcelClear = now.getTime();
+    }
+}
+
+export async function clearExcelCacheHelper
+({lastExcelClear, now, cachePath, environment}: {lastExcelClear: number | null, now: Date, cachePath: string, environment: "production" | "development" | "staging" | "test"}): Promise<boolean> {
+    if (environment === "development") {
+        return false;
+    }
+
+    const hour = now.getHours();
+
+    // between 3 and 6 AM
+    if(hour < 3 || hour > 5) {
+        return false;
+    }
+
+    // only run once each day
+    if(lastExcelClear !== null && lastExcelClear + msIn22Hours > now.getTime()) {
+        return false;
+    }
+
+    const currentYear = now.getFullYear();
+    const currentMonth = now.getMonth();
+    const currentDay = now.getDate();
+
+    const maxDaysInCache = 2;
+
+    const dateLimit = new Date(currentYear, currentMonth, currentDay - maxDaysInCache, 0,0,0,0);
+
+    const files = await fs.readdir(cachePath, {withFileTypes: true});
+
+    for (const file of files) {
+        if (file.isDirectory()) {
+            try {
+                const date = getDateFromDirectoryName(file.name);
+                const shouldDelete = date < dateLimit;
+
+                if(shouldDelete) {
+                    const path = file.path + '/' + file.name;
```

Comment:
file.path is deprecated in Node: https://nodejs.org/api/fs.html#direntpath

```suggestion
                    const path = file.parentPath + '/' + file.name;
```